### ISSUE TITLE
Current history state sync error

### ISSUE THREAD

**Issue by dannon** [galaxyproject/galaxy#14173]
**Describe the bug**
Current history can get out of sync.  This video displays the issue -- on page load we get history A (single step random lines), and then after a few seconds the update swaps to history B (unnamed history).

https://user-images.githubusercontent.com/155398/175325009-ecaa95fb-1c3f-4a46-afe6-740dc4e01580.mp4

Looking at the requests, we see another issue -- `current_history_json` is reporting results and id from history A, but the requests to contents/view are coming from both:

![image](https://user-images.githubusercontent.com/155398/175325470-474855a7-a246-475a-8df7-7b29fa0a44c8.png)

**Comment by davelopez** [comment]
Ic can reproduce this out-of-sync behavior consistently if we delete a history. 
Hopefully, that helps to track this issue down.

## Steps
With at least 2 histories in your account:
- Delete History A
- Observe the current history flickers between both histories and finally History A (the deleted one) is shown.
- Refresh the page
- The current history is really History B


![history_out_of_sync](https://user-images.githubusercontent.com/46503462/177530744-e20117a9-51a8-4cec-a167-c169faef257d.gif)

**Comment by davelopez** [comment]
I added some traces while debugging another issue with history updates and I found more info about this one, so decided to post it here.
To add some context, my (real) current history is `96d9e11f37f34b29` (named `Public History`), if I do a page refresh I see the following log:

```log
...
[truncated]
...
index.js?9c8a:12 Analysis custom page setup
historyStore.js?d1ca:52 STORE currentHistoryId null NOT FOUND in histories size=0
queries.js?a2f0:57 LOADING ALL histories from server
[Violation] 'setInterval' handler took 191ms
syncConfigToGalaxy.js?9619:14 syncConfigToGalaxy complete
url.js?4b90:7 Requesting data from:  api/histories/96d9e11f37f34b29/contents?limit=1000&q=update_time-ge&qv=2022-07-19T12%3A22%3A30.177Z&v=dev&view=detailed
historyStore.js?d1ca:52 STORE currentHistoryId null NOT FOUND in histories size=15
queries.js?a2f0:69 LOADING history 7b668ee810f6cf46 (detailed) from server
url.js?4b90:7 Requesting data from:  api/histories/7b668ee810f6cf46/contents?v=dev&order=hid&offset=0&limit=100&q=deleted&qv=false&q=visible&qv=true
historyStore.js?d1ca:30 STORE setHistory ID=96d9e11f37f34b29 with name='Public History'
historyStore.js?d1ca:52 STORE currentHistoryId null NOT FOUND in histories size=15
HistoryPanel.vue?488a:228 CURRENT HISTORY CHANGED from 7b668ee810f6cf46 to 7b668ee810f6cf46 -> Changes: {}
historyStore.js?d1ca:30 STORE setHistory ID=96d9e11f37f34b29 with name='Public History'
historyStore.js?d1ca:26 STORE setCurrentHistoryId ID=96d9e11f37f34b29
historyStore.js?d1ca:49 STORE currentHistoryId 96d9e11f37f34b29 FOUND in histories size=15
queries.js?a2f0:69 LOADING history 96d9e11f37f34b29 (detailed) from server
HistoryPanel.vue?488a:228 CURRENT HISTORY CHANGED from 7b668ee810f6cf46 to 96d9e11f37f34b29 -> Changes: {id: '7b668ee810f6cf46', importable: null, size: null, contents_url: null, genome_build: null, …}
historyStore.js?d1ca:30 STORE setHistory ID=7b668ee810f6cf46 with name='Different history'
queries.js?a2f0:57 LOADING ALL histories from server
historyStore.js?d1ca:30 STORE setHistory ID=96d9e11f37f34b29 with name='Public History'
HistoryPanel.vue?488a:228 CURRENT HISTORY CHANGED from 96d9e11f37f34b29 to 96d9e11f37f34b29 -> Changes: {url: 'history?id=96d9e11f37f34b29', importable: true, contents_url: 'history_contents?history_id=96d9e11f37f34b29', genome_build: '?', user_id: 'f2db41e1fa331b3e', …}
historyStore.js?d1ca:49 STORE currentHistoryId 96d9e11f37f34b29 FOUND in histories size=15
HistoryPanel.vue?488a:228 CURRENT HISTORY CHANGED from 96d9e11f37f34b29 to 96d9e11f37f34b29 -> Changes: {contents_active: {…}, size: 32079}
url.js?4b90:7 Requesting data from:  api/histories/96d9e11f37f34b29/contents?v=dev&order=hid&offset=0&limit=100&q=deleted&qv=false&q=visible&qv=true
```

The first requests are for a `Different History` (`7b668ee810f6cf46`) which happens to be the *first history in my histories list*. So it look like here is where it is getting the *first history* instead of my current history:

https://github.com/galaxyproject/galaxy/blob/d49c72b7b127ed77f653cecad2584edbb9f5a738/client/src/store/historyStore/historyStore.js#L46

Since the initial value for currentHistoryId in the store is null, we are falling back to the first history in our histories list after they have been loaded, and, at some point after, we set the setCurrentHistoryId to the real one.

Also notice that we are loading ALL histories twice for each page refresh...

**Comment by dannon** [comment]
Fixed with #14376.
